src/open_clip/coca_model.py

```diff
@@ -0,0 +1,209 @@
+from typing import Optional
+
+import torch
+from torch import nn
+from torch.nn import functional as F
+import numpy as np
+from dataclasses import dataclass
+
+from .transformer import (
+    LayerNormFp32,
+    LayerNorm,
+    QuickGELU,
+    MultimodalTransformer,
+)
+from .model import CLIPTextCfg, CLIPVisionCfg, _build_vision_tower, _build_text_tower
+from .generation_utils import top_a, top_k, top_p
+
+
+@dataclass
+class MultimodalCfg(CLIPTextCfg):
+    mlp_ratio: int = 4
+    dim_head: int = 64
+    heads: int = 8
+    n_queries: int = 256
+    attn_pooler_heads: int = 8
+    latent_dim: int = 512
+
+
+class CoCaEncoderDecoder(nn.Module):
+    def __init__(self, encoder, decoder) -> None:
```

Comment:
Is there are reason for this module to exist? vs a text_encoder and tex_decoder in CoCa model? having modules organized like this (two modules in a class with no forward that exercises) is atypical and breaks some assumptions for profilers, etc

Reply:
FYI, checkpoints can easily be remapped, so existing checkpoints is not a reason not to make changes right now

Reply:
More thoughts, to keep it a bit closer to the Clip model on the encoder side 

```
self.text = _build_text_encoder()
self.visual = _build_vision_tower()
self.decoder = _build_multimodal_decoder() # or self.multimodal_decoder or .text_decoder?
self.decoder_norm = 
self.decoder_logits =
```